lightning/src/chain/keysinterface.rs

```diff
@@ -233,13 +233,16 @@ pub trait ChannelKeys : Send+Clone {
 	// TODO: Document the things someone using this interface should enforce before signing.
 	fn sign_counterparty_commitment<T: secp256k1::Signing + secp256k1::Verification>(&self, commitment_tx: &CommitmentTransaction, secp_ctx: &Secp256k1<T>) -> Result<(Signature, Vec<Signature>), ()>;
 
-	/// Create a signature for a holder's commitment transaction. This will only ever be called with
-	/// the same commitment_tx (or a copy thereof), though there are currently no guarantees
-	/// that it will not be called multiple times.
+	/// Create a signatures for a holder's commitment transaction and its claiming HTLC transactions.
+	/// This will only ever be called with a non-revoked commitment_tx.  This may be the latest
+	/// tx or it may be the second latest if not revoked and some watchtower/secondary
+	/// ChannelMonitor decided to broadcast before it had been updated to the latest.
+	/// This may be called multiple times.
```

Comment:
nit: maybe mention that it can be called multiple times for the same commitment_tx - its obviously called regularly for different ones.

Reply:
> called multiple times for the same commitment_tx - its obviously called regularly for different ones.

I agree with the first part and will adjust the doc, but I don't understand what you mean by the second part.  It is *not* called regularly - it's only called at force-close which happens rarely, right?